xbmc/video/VideoDatabase.cpp

```diff
@@ -3034,28 +3046,89 @@ bool CVideoDatabase::SetFileForEpisode(const std::string& fileAndPath, int idEpi
   return false;
 }
 
-bool CVideoDatabase::SetFileForMovie(const std::string& fileAndPath, int idMovie, int idFile)
+bool CVideoDatabase::SetFileForMovie(const std::string& fileAndPath, int idMovie)
 {
+  const int idFile{AddFile(fileAndPath)};
+  if (idFile < 0)
+    return false;
+
+  const int oldIdFile{GetFileIdByMovie(idMovie)};
+  if (oldIdFile < 0)
+    return false;
+
```

Comment:
As mentioned in other comment, the original file id needs to come from the caller, because GetFileIdByMovie can only retrieve the file of the default version of a movie.